pkg/reconciler/pipelinerun/cancel_test.go

```diff
@@ -104,46 +110,189 @@ func TestCancelPipelineRun(t *testing.T) {
 				Status: v1beta1.PipelineRunSpecStatusCancelledDeprecated,
 			},
 		},
+	}, {
+		name: "child-references",
+		pipelineRun: &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pipeline-run-cancelled"},
+			Spec: v1beta1.PipelineRunSpec{
+				Status: v1beta1.PipelineRunSpecStatusCancelled,
+			},
+			Status: v1beta1.PipelineRunStatus{PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+				ChildReferences: []v1beta1.ChildStatusReference{
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun"},
+						Name:             "t1",
+						PipelineTaskName: "task-1",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun"},
+						Name:             "t2",
+						PipelineTaskName: "task-2",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "Run"},
+						Name:             "r1",
+						PipelineTaskName: "run-1",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "Run"},
+						Name:             "r2",
+						PipelineTaskName: "run-2",
+					},
+				},
+				// If ChildReferences is not empty, TaskRuns and Runs should be ignored.
+				TaskRuns: map[string]*v1beta1.PipelineRunTaskRunStatus{
+					"t3": {PipelineTaskName: "task-3"},
+					"t4": {PipelineTaskName: "task-4"},
+				},
+				Runs: map[string]*v1beta1.PipelineRunRunStatus{
+					"r3": {PipelineTaskName: "run-3"},
+					"r4": {PipelineTaskName: "run-4"},
+				},
+			}},
+		},
+		taskRuns: []*v1beta1.TaskRun{
+			{ObjectMeta: metav1.ObjectMeta{Name: "t1"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "t2"}},
+		},
+		runs: []*v1alpha1.Run{
+			{ObjectMeta: metav1.ObjectMeta{Name: "r1"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "r2"}},
+		},
+		ignoredTaskRuns: []*v1beta1.TaskRun{
+			{ObjectMeta: metav1.ObjectMeta{Name: "t3"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "t4"}},
+		},
+		ignoredRuns: []*v1alpha1.Run{
+			{ObjectMeta: metav1.ObjectMeta{Name: "r3"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "r4"}},
+		},
+	}, {
+		name: "unknown-kind-on-child-references",
+		pipelineRun: &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pipeline-run-cancelled"},
+			Spec: v1beta1.PipelineRunSpec{
+				Status: v1beta1.PipelineRunSpecStatusCancelled,
+			},
+			Status: v1beta1.PipelineRunStatus{PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+				ChildReferences: []v1beta1.ChildStatusReference{{
+					TypeMeta:         runtime.TypeMeta{Kind: "InvalidKind"},
+					Name:             "t1",
+					PipelineTaskName: "task-1",
+				}},
+			}},
+		},
+		expectedErr: errors.New("error(s) from cancelling TaskRun(s) from PipelineRun test-pipeline-run-cancelled: unknown or unsupported kind `InvalidKind` for cancellation"),
```

Comment:
any way to avoid asserting the exact error message? either a regex or just a boolean wantErr

Reply:
Probably - lemme create a followup to do this systematically, 'cos I've seen this sort of thing in a bunch of places.

Reply:
I did not, I will do that after a call, and will update here as well.

Reply:
I forgot again to open the issue! Doing so right now.